### DISCUSSION TITLE
how do i get the auto recorded demo out of the container?

### DISCUSSION THREAD

**Discussion by BabyFoxy** [timche/docker-csgo#17] [Q&A] [answered]
I have enabled record=1:1 in PUGSETUP_SETUPOPTIONS, just wondering, how I can get the demo out from the container when the recording is finished? TIA

**Comment by timche** [comment, marked as answer]
The demos files are stored in the `/home/csgo/server` bound volume. If you need access to the volume, you should use bind mounts: https://docs.docker.com/storage/bind-mounts/ so the volume is bound to a directory in e.g. your home folder. After that you can set up an FTP server or web server with directory listing with Docker to download the demo files.

**Comment by BabyFoxy** [comment]
I was able to mount the volume, however, after waiting for the updating, I have got an error msg said could not be written due to permission, any idea what went wrong? I was mount with RW permission already

**Reply by timche** [reply]
Can you please share the log and the command to run the container?

**Comment by BabyFoxy** [comment]
```
Update state (0x101) committing, progress: 93.31 (26647851797 / 28559647439)
Success! App '740' fully installed.
+ set +x
> Done
touch: cannot touch '/home/csgo/server/installed.lock': Permission denied
> Installing server ...
+ /home/csgo/Steam/steamcmd.sh +login anonymous +force_install_dir /home/csgo/server +app_update 740 validate +quit
WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
Redirecting stderr to '/home/csgo/Steam/logs/stderr.txt'
```

I was running it with Synology DSM 7.0's Docker, not too sure how it was running ...

**Reply by timche** [reply]
Can you please share your `docker run` command?

**Reply by BabyFoxy** [reply]
emm.... not too sure how it was running, as everything was done via GUI in Synology DSM

**Reply by timche** [reply]
You should probably look into your Synology DSM manual on how to do bind mounting. I assume the path it mounted to doesn't exist or has no permissions.

**Reply by BabyFoxy** [reply]
ok, turns out it was permission on the DSM folder, had to set permission to allow "everyone" be able to read and write, although from security point of view probably not ideal but I can live with it.

Thanks for the help
